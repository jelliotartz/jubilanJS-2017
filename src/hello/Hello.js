import React, { Component } from 'react';

class Hello extends Component {
  constructor(props) {
    super(props);
    this.state = {
      greeting: 'Hiya'
    };
  }

  render() {
    return (
      <div className="hello">
        <h1>{this.state.greeting}</h1>
        <h2>{this.props.friend}</h2>
        <p>Congratulations! You created a React Component!</p>
      </div>
    );
  }
}

Hello.defaultProps = {
  friend: 'Bud :)'
}

export default Hello;
