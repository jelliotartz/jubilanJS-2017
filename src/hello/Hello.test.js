import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Hello from './Hello';

describe('Hello World:', function () {
  it('renders without exploding', () => {
    expect(shallow(<Hello />)).toHaveLength(1);
  });

  it('should render with default text', function () {
    const component = shallow(<Hello />);

    expect(component).toIncludeText('Hiya');
    expect(component).toIncludeText('Bud :)');
  });

  it('should render with our props', function () {
    const component = shallow(
      <Hello friend="Fred"/>
    );

    expect(component).toIncludeText('Hiya');
    expect(component).toIncludeText('Fred');
    expect(component).not.toIncludeText('Bud :)');
  })

  it('should render to match the snapshot', function () {
    const component = renderer.create(
      <Hello friend="Fritz"/>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

});
