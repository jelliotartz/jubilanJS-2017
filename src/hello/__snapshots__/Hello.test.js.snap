exports[`Hello World: should render to match the snapshot 1`] = `
<div
  className="hello">
  <h1>
    Hiya
  </h1>
  <h2>
    Fritz
  </h2>
  <p>
    Congratulations! You created a React Component!
  </p>
</div>
`;
